File: shoring/flywheel/src/test/java/com/panopset/tests/transformer/TransformerTest.java
package com.panopset.tests.transformer;

import org.junit.jupiter.api.Assertions;

import com.panopset.tests.TestDirectory;

public abstract class TransformerTest implements TestDirectory  {

	
	protected abstract ResultsDataSupplier createResultsDataSupplier();
	
	private ResultsDataSupplier rds;
	
	private ResultsDataSupplier getResultsDataSupplier() {
		if (rds == null) {
			rds = createResultsDataSupplier();
		}
		return rds;
	}
	
//	public TransformerTest(final String expectedResults) {
//		this(new ResultsDataSupplier() {
//			
//			@Override
//			public String getExpectedResults() {
//				return expectedResults;
//			}
//		});
//	}
//	
//	public TransformerTest(final File expectedResultsFile) {
//		this(new ResultsDataSupplier() {
//			
//			@Override
//			public String getExpectedResults() {
//				return Fileop.readTextFile(expectedResultsFile);
//			}
//		});
//	}
	
	public void test() {
		init();
		Assertions.assertEquals(getResultsDataSupplier().getExpectedResults(), getResultsDataSupplier().getActualResults());
	}
	
	protected void init() {
		
	}

}

File: code/shoring/flywheel/src/test/kotlin/com/panopset/tests/transformer/TransformerTest.kt
package com.panopset.tests.transformer

import org.junit.jupiter.api.Assertions

abstract class TransformerTest {
    protected abstract fun createResultsDataSupplier(): ResultsDataSupplier
    private val resultsDataSupplier: ResultsDataSupplier
        get() = createResultsDataSupplier()



    //	public TransformerTest(final String expectedResults) {
    //		this(new ResultsDataSupplier() {
    //
    //			@Override
    //			public String getExpectedResults() {
    //				return expectedResults;
    //			}
    //		});
    //	}
    //
    //	public TransformerTest(final File expectedResultsFile) {
    //		this(new ResultsDataSupplier() {
    //
    //			@Override
    //			public String getExpectedResults() {
    //				return Fileop.readTextFile(expectedResultsFile);
    //			}
    //		});
    //	}
    fun test() {
        Assertions.assertEquals(resultsDataSupplier.expectedResults, resultsDataSupplier.actualResults)
    }
}
